File: 240304/두 가지로 열리는 자물쇠/a-two-way-lock.py
n = int(input())
a, b, c = tuple(map(int, input().split()))
a2, b2, c2 = tuple(map(int, input().split()))


cnt = 0
for i in range(1, n + 1):
    for j in range(1, n + 1):
        for k in range(1, n + 1):
            if (abs(a - i) <= 2 or abs(a - i) >= n - 2) and (abs(b - j) <= 2 or abs(b - j) >= n - 2) and \
               (abs(c - k) <= 2 or abs(c - k) >= n - 2):
                cnt += 1
			
            elif (abs(a2 - i) <= 2 or abs(a2 - i) >= n - 2) and (abs(b2 - j) <= 2 or abs(b2 - j) >= n - 2) and \
               (abs(c2 - k) <= 2 or abs(c2 - k) >= n - 2):
                cnt += 1

print(cnt)

File: 260112/두 가지로 열리는 자물쇠/a-two-way-lock.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const [a, b, c] = input[1].split(' ').map(Number);
const [a2, b2, c2] = input[2].split(' ').map(Number);

// Please Write your code here.
function isValid(arr1, arr2){
    for(let i=0; i<3; i++){
        const num1 = arr1[i], num2 = arr2[i];
        // console.log(`num1: ${num1}, num2: ${num2}`)
        if((num1===1 && num2===9) || (num1===2 && num2===8) || (num1===9 && num2===1) || (num1===2 && num2===8)){
            continue;
        }
        if(Math.abs(num1-num2)>2){
            return false;
        }
    }
    return true;
}

let ans = 0;

for(let i=1; i<=n; i++){
    for(let j=1; j<=n; j++){
        for(let k=1; k<=n; k++){
            // console.log(`i: ${i}, j: ${j}, k: ${k}`)
            if(isValid([i,j,k], [a,b,c]) || isValid([i,j,k], [a2,b2,c2])){
                ans++;
            }
        }
    }
}

console.log(ans);